pages/cartPage.js

```diff
@@ -0,0 +1,296 @@
+import Head from 'next/head';
+import React from 'react';
+import Header from '../components/Header.js';
+import Footer from '../components/Footer.js';
+import Link from 'next/link';
+import cookie from 'js-cookie';
+import nextCookies from 'next-cookies';
+import RemoveFromCart from '../components/RemoveFromCart.tsx';
+import AddOneItem from '../components/AddOneItem.tsx';
+import ReduceOneItem from '../components/ReduceOneItem.tsx';
+
+export function totalSum(itemsInCart) {
+  return itemsInCart.length !== 0
+    ? itemsInCart.reduce((acc, cur) => {
+        return acc + cur.price;
+      }, 0)
+    : 0;
```

Comment:
Code readability (not such a huge deal):

If you're in a ternary with a long value like this, it may be worth it to consider putting the short value before the long one:

```js
  return itemsInCart.length === 0
    ? 0
    : itemsInCart.reduce((acc, cur) => {
        return acc + cur.price;
      }, 0);
```

This way, you can see right away what happens if `length` is `0`, and you don't need to go searching for that `:`.

Another way of doing it (maybe even nicer?) is to do an early return after the `if`:

```js
if (itemsInCart.length === 0) return 0;

return itemsInCart.reduce((acc, cur) => {
  return acc + cur.price;
}, 0);
```

Then you have even less noise from the ternary.

Reply:
Oh wait, even simpler - no check for `.length`:

```js
return itemsInCart.reduce((acc, cur) => {
  return acc + cur.price;
}, 0);
```

The `reduce` starts with the value `0`.